tests/end-to-end/backup-restore-test/backupe2e/asset-store/asset-store.go

```diff
@@ -0,0 +1,80 @@
+package backupe2eAssetStore
+
+import (
+	"github.com/google/uuid"
+	"github.com/kyma-project/kyma/tests/end-to-end/backup-restore-test/backupe2e/asset-store/testsuite"
+	restclient "k8s.io/client-go/rest"
+
+	"github.com/kyma-project/kyma/tests/end-to-end/backup-restore-test/utils/config"
+	. "github.com/smartystreets/goconvey/convey"
+	"testing"
+)
+
+type assetStoreTest struct {
+	uuid       string
+	restConfig *restclient.Config
+	testSuite  *testsuite.TestSuite
+	t          *testing.T
+}
+
+func NewAssetStoreTest(t *testing.T) (*assetStoreTest, error) {
+	restConfig, err := config.NewRestClientConfig()
+	if err != nil {
+		return &assetStoreTest{}, err
+	}
+
+	return &assetStoreTest{
+		restConfig: restConfig,
+		testSuite:  nil,
+		uuid:       uuid.New().String(),
+		t:          t,
+	}, nil
+}
+
+func (a *assetStoreTest) CreateResources(namespace string) {
+	testSuite, err := testsuite.New(a.restConfig, namespace, a.t)
+	So(err, ShouldBeNil)
+	a.setTestSuite(testSuite)
+
+	err = a.testSuite.CreateBuckets()
+	So(err, ShouldBeNil)
+
+	err = a.testSuite.CreateAssets()
+	So(err, ShouldBeNil)
+}
+
+func (a *assetStoreTest) TestResources(namespace string) {
+	err := a.testSuite.WaitForBucketsReady()
+	So(err, ShouldBeNil)
+
+	err = a.testSuite.WaitForAssetsReady()
+	So(err, ShouldBeNil)
+}
+
+func (a *assetStoreTest) DeleteResources(namespace string) error {
+	err := a.testSuite.DeleteClusterBucket()
+	if err != nil {
```

Comment:
That's what I was writing about unifying approach. If the other methods use GoConvey for validation (and error handling), then you should use it here as well.